Userland/Libraries/LibCrypto/Hash/SHA2.cpp

```diff
@@ -56,14 +69,67 @@ inline void SHA256::transform(u8 const* data)
         a = temp0 + temp1;
     }
 
-    m_state[0] += a;
-    m_state[1] += b;
-    m_state[2] += c;
-    m_state[3] += d;
-    m_state[4] += e;
-    m_state[5] += f;
-    m_state[6] += g;
-    m_state[7] += h;
+    state[0] += a;
+    state[1] += b;
+    state[2] += c;
+    state[3] += d;
+    state[4] += e;
+    state[5] += f;
+    state[6] += g;
+    state[7] += h;
+}
+
+#if (ARCH(I386) || ARCH(X86_64)) && !defined KERNEL && defined(AK_COMPILER_GCC)
+
+SHA256_ATTRIBUTE_TARGET_X86 static void SHA256_transform_impl(u32 (&state)[8], u8 const (&data)[64])
+{
+    using AK::SIMD::i32x4;
+
+    i32x4 states[2] {};
+    states[0] = AK::SIMD::load_unaligned<i32x4>(&state[0]);
+    states[1] = AK::SIMD::load_unaligned<i32x4>(&state[4]);
+    auto tmp = i32x4 { states[0][1], states[0][0], states[0][3], states[0][2] };
+    states[1] = i32x4 { states[1][3], states[1][2], states[1][1], states[1][0] };
+    states[0] = i32x4 { states[1][2], states[1][3], tmp[0], tmp[1] };
+    states[1] = i32x4 { states[1][0], states[1][1], tmp[2], tmp[3] };
+
+    i32x4 msgs[4] {};
+    i32x4 old[2] { states[0], states[1] };
+    for (int i_msg = 0; i_msg != 16; ++i_msg) {
+        i32x4 msg {};
+        if (i_msg < 4) {
+            msgs[(i_msg + 0) % 4] = AK::SIMD::load_unaligned<i32x4>(&data[i_msg * 16]);
+            msgs[(i_msg + 0) % 4] = AK::SIMD::elementwise_byte_reverse(msgs[(i_msg + 0) % 4]);
+            tmp = AK::SIMD::load_unaligned<i32x4>(&SHA256Constants::RoundConstants[i_msg * 4]);
+            msg = msgs[(i_msg + 0) % 4] + tmp;
```

Comment:
as this is only executed in the first 4 iterations the `% 4` can be elided

Reply:
Fixed.